tests/integration/src/codegen/intrinsics/mem.rs

```diff
@@ -731,3 +731,99 @@ fn store_u8() {
         _ => panic!("Unexpected test result: {res:?}"),
     }
 }
+
+#[test]
+fn store_unaligned_u32() {
+    // Use the start of the 17th page (1 page after the 16 pages reserved for the Rust stack)
+    let write_to = 17 * 2u32.pow(16);
+    let write_val = 0xddccbbaa_u32; // Little-endian bytes will be [AA BB CC DD].
+
+    // Generate a `test` module with `main` function that stores to a u32 offset.
+    let signature = Signature::new(
+        [AbiParam::new(Type::U32)],
+        [AbiParam::new(Type::U32)], // Return u32 to satisfy test infrastructure
+    );
+
+    // Compile once outside the test loop
+    let (package, context) = compile_test_module(signature, |builder| {
+        let block = builder.current_block();
+        let idx_val = block.borrow().arguments()[0] as ValueRef;
+
+        // Set base pointer, add argument offset to it.
+        let base_addr = builder.u32(write_to, SourceSpan::default());
+        let write_addr = builder.add(base_addr, idx_val, SourceSpan::default()).unwrap();
+        let ptr = builder
+            .inttoptr(write_addr, Type::from(PointerType::new(Type::U32)), SourceSpan::default())
+            .unwrap();
+
+        // Store test value to pointer.
+        let write_val = builder.u32(write_val, SourceSpan::default());
+        builder.store(ptr, write_val, SourceSpan::default()).unwrap();
+
+        // Return a constant to satisfy test infrastructure
+        let result = builder.u32(1, SourceSpan::default());
+        builder.ret(Some(result), SourceSpan::default()).unwrap();
+    });
+
+    fn run_test(
+        package: &miden_mast_package::Package,
+        context: &std::rc::Rc<midenc_hir::Context>,
+        write_to: u32,
+        idx: u32,
```

Comment:
`idx` -> "offset"? Also, the first three arguments can be removed since we're passing the same values on every call.